pkg/plugins/golang/v3/api.go

```diff
@@ -210,6 +211,8 @@ func (p *createAPISubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
 
 	// Create the actual resource from the resource options
 	res := p.resource.NewResource(p.config, p.doResource)
+	res.API = p.doResource
+	res.Controller = p.doController
```

Comment:
I think that it will be cleaner if you:
```suggestion
	res := p.resource.NewResource(p.config, p.doResource, p.doController)
```
And then set those two fields in resource creation. Right now you can't do that because you dont want to set them for `v2` project files for backwards compatibility. But could we leave a comment about this:
```suggestion
	// TODO: API & Controller should be set inside `Options.NewResource` once v2 is no longer supported.
	res := p.resource.NewResource(p.config, p.doResource)
	res.API = p.doResource
	res.Controller = p.doController
```

Reply:
Really thank you for your input. I liked it and i addressed your suggestion. 
See that for we do not change the current behaviour for v2 we can only check the if is v2 version.